core-library/src/main/java/org/silverpeas/core/io/media/image/imagemagick/Im4javaImageTool.java

```diff
@@ -54,6 +66,23 @@ public boolean isActivated() {
     return Im4javaManager.isActivated();
   }
 
+  @Override
+  public String[] getImageInfo(final File source, final ImageInfoType... infoTypes) throws Exception {
+    return identify(source,
+        Arrays.stream(infoTypes).map(ImageInfoType::getImOption).toArray(String[]::new));
+  }
+
+  private String[] identify(final File source, final String... infoTypes) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00112)